lib/report_formatter/timeline.rb

```diff
@@ -214,68 +221,15 @@ def tl_event(tl_xml, row, col)
-        elsif co == "ext_management_system.name" && rec.ext_management_system && !rec.ext_management_system.id.nil?
-          provider_id = rec.ext_management_system.id
-          if ems_cloud
-            # restful route is used for cloud provider unlike infrastructure provider
-            val = "&lt;a href='/ems_cloud/#{provider_id}'&gt;#{row[co]}&lt;/a&gt;"
-          elsif ems_container
-            val = "&lt;a href='/ems_container/show/#{to_cid(provider_id)}'&gt;#{row[co]}&lt;/a&gt;"
-          else
-            val = "&lt;a href='/ems_infra/show/#{to_cid(provider_id)}'&gt;#{row[co]}&lt;/a&gt;"
-          end
-        elsif co == "availability_zone.name" && !rec.availability_zone_id.nil?
-          val = "&lt;a href='/availability_zone/show/#{to_cid(rec.availability_zone_id)}'&gt;#{row[co]}&lt;/a&gt;"
-        elsif co == "container_node_name"
-          if rec.container_node_id
-            val = "<a href='/container_node/show/#{to_cid(rec.container_node_id.id)}'>#{row[co]}</a>"
-          end
-        elsif co == "container_group_name"
-          if rec.container_group_id
-            val = "<a href='/container_group/show/#{to_cid(rec.container_group.id)}'>#{row[co]}</a>"
-          end
-        elsif co == "container_name"
-          if rec.container_id
-            val = "<a href='/container/tree_select/?id=cnt-#{to_cid(rec.container_id)}'>#{row[co]}</a>;"
-          end
-        elsif co == "container_replicator_name" && rec.container_replicator_id
-          if rec.container_replicator_id
-            val = "<a href='/container_replicator/show/#{to_cid(rec.container_replicator_id)}'>#{row[co]}</a>"
-          end
-        elsif mri.db == "BottleneckEvent" && co == "resource_name"
-          case rec.resource_type
-          when "ExtManagementSystem"
-            db = ems_cloud ? "ems_cloud" : "ems_infra"
-          else
-            db = rec.resource_type.underscore
-          end
-          val = "&lt;a href='/#{db}/show/#{to_cid(rec.resource_id)}'&gt;#{rec.resource_name}&lt;/a&gt;"
-        else  # Not a link field, just put in the text
-          # START of TIMELINE TIMEZONE Code
-          if row[co].kind_of?(Time)
-            val = format_timezone(row[co], tz, "gtl")
-          elsif TIMELINE_TIME_COLUMNS.include?(co)
-            val = format_timezone(Time.parse(row[co].to_s), tz, "gtl")
-          else
-            val = row[co].to_s
-          end
-          # END of TIMELINE TIMEZONE Code
-          #           e_text += row[co].to_s
-        end
+        href_text = if row[co].kind_of?(Time)
+                      format_timezone(row[co], tz, "gtl")
+                    elsif TIMELINE_TIME_COLUMNS.include?(co)
+                      format_timezone(Time.parse(row[co].to_s).utc, tz, "gtl")
+                    else
+                      row[co].to_s
+                    end
+
+        val = tl_href(co, rec, href_text, ems_cloud, ems_container) || ''
```

Comment:
> > +        val = tl_href(co, rec, href_text, ems_cloud, ems_container) ||
''
>
> Sure || '' is needed?
> #{nil} == nil.to_s == ''
>
tl_href() usually returns strings, no?
Wouldnt it be cleaner to make it return "" rather than nil?